asllib/Parser.mly

```diff
@@ -90,96 +90,12 @@ let e_binop (e1, op, e2) =
-%token PRINT RBRACE RBRACKET RDIV REAL RECORD REPEAT RETURN RPAR STAR_COLON
-%token SEMI_COLON SETTER SHL SHR SLICING STRING SUBTYPES THEN THROW TO TRY TYPE
-%token UNKNOWN UNREACHABLE UNTIL VAR WHEN WHERE WHILE WITH
-
-%token ARROBASE_LOOPLIMIT
-%token <string> IDENTIFIER STRING_LIT
-%token <Bitvector.mask> MASK_LIT
-%token <Bitvector.t> BITVECTOR_LIT
-%token <Z.t> INT_LIT
-%token <Q.t> REAL_LIT
-%token <bool> BOOL_LIT
-
-
-(* ------------------------------------------------------------------------
-
-                           Associativity and priority
-
-  ------------------------------------------------------------------------- *)
-
-(*
-   This section on associativity uses menhir associativity and priority
-   features. Internally, it is used by menhir to resolve some conflicts that
-   could arrise from different conflicting expressions, e.g. [3 + 4 + 5].
-
-   For a quick intro, menhir assigns a priority level to tokens that have a
-   [left], [right], or [nonassoc] declaration in the order in which they are
-   declared. For example, here [PLUS]'s associativity is declared before [MUL]
-   so [3 + 4 * 5] will be parsed as [3 + (4 * 5)].
-
-   Associativity is straigh-forward.
-
-   Priority declarations that follow are created because of the fusion of
-   multiple recursive bnf rules into one, e.g. [expr] is the fusion of [expr]
-   and many others such as [cexpr].
-   The rule tree that I am translating here into priority rules is the
-   following:
-
-     expr <-----------------------|IF|----------------------< cexpr
-     cexpr <----|binop_boolean, checked_type_constraint|---<  cexpr_cmp
-     cexpr_cmp <-----------|binop_comparison|---------------< cexpr_add_sub
-     cexpr_add_sub <------|binop_add_sub_logic|-------------< cexpr_mul_div
-     cexpr_mul_div <------|binop_mul_div_shift|-------------< cexpr_pow
-     cepxr_pow <---------------|binop_pow|------------------< bexpr
-     bexpr <---------------------|unop|---------------------< expr_term
-     expr_term <------------------|IN|----------------------< expr_atom
-     expr_atom <-----------|DOT, brackets, ...|-------------< expr
-
-
-  Note that the token MINUS has two different precedence: one for when it is a
-  binary operator, in that case it has the same precedence as PLUS, and one for
-  when it is a unary operator, in which case it has the same precendence as
-  NOT.
-*)
-
-(* IF *)
-%nonassoc ELSE
-
-(* binop_boolean, checked_type_constraint *)
-%left BOR BAND IMPL BEQ AS
-
-(* binop_comparison *)
-%left EQ_OP NEQ
-%nonassoc GT GEQ LT LEQ
-
-(* binop_add_sub_logic *)
-%left PLUS MINUS OR EOR AND
-
-(* binop_mul_div_shift *)
-%left MUL DIV DIVRM RDIV MOD SHL SHR
-
-(* binop_pow *)
-%left POW CONCAT
-
-(* unop: NOT, BNOT, MINUS *)
-%nonassoc UNOPS
-
-(* IN *)
-%nonassoc IN
-
-(* DOT, brackets, etc. *)
-%left DOT LBRACKET
+%parameter<Config : sig
+    val allow_no_end_semicolon : bool
+end>
```

Comment:
Sadly I don't think this is possible :/ Quoting from the menhir docs:
```
... Last, %parameter declarations take effect before %{ … %}, %token, %type, or %start declarations are considered, so that the module name uid introduced by a %parameter declaration is effectively in scope in all %{ … %}, %token, %type, or %start declarations, regardless of whether they precede or follow the %parameter declaration. This means, in particular, that the side effects of an OCaml header are observed only when the functor is applied, not when it is defined.
```

What I could do is move the parameter just above the header such that it's a bit neater.

Reply:
Making it a `.mli` file throws an error since no `.ml` file actually exists to implement it, but `ParserConfig.ml` works.